client/app/pages/queries/source-view.js

```diff
@@ -55,6 +58,17 @@ function QuerySourceCtrl(
   $scope.updateQuery = debounce(
     newQueryText => $scope.$apply(() => {
       $scope.query.query = newQueryText;
+      $scope.dryRunDirty = true;
+    }),
+  );
+
+  $scope.updateQueryDone = debounce(
```

Comment:
that's for identifying when the dry run has completed ?

Reply:
No, that's identifying when you're done typing in the editor for a few seconds. 
This is a trigger for the dry run to start rather than getting a response from google.